chainer/links/connection/linear.py

```diff
@@ -99,6 +99,7 @@ def __init__(self, in_size, out_size=None, nobias=False,
 
         if out_size is None:
             in_size, out_size = None, in_size
+        self.in_size = in_size
```

Comment:
How about `self._in_size_spec`?  Leaving `self.in_size` `None` after `_initialize_params` is OK for printing, while it could be confusing for other purposes.

Reply:
No.  `L.Linear` does not.  I guess there is a code where `L.Linear` is inherited and `self.in_size` and `_initialize_params` are patched to do so.

Reply:
You are right.  I was just surprised at that an attribute is added for printing.  Comparing with other links, it's quite natural to have `Linear.in_size` attribute.